### ISSUE TITLE
Errors when preparing environment in BundlesTest are invisibly swallowed 

### ISSUE THREAD

**Issue by victornoel** [zerocracy/farm#1739]
I am trying to solve #1162 and I got blocked because an error during the processing of the `claims.xml` file of a bundle got swallowed and never shown to me.

I had to realise it happened by going through debug.

The situation was the following:
- trying to make `publish_project` bundle work
- the first claim of the claims.xml is a "Make Payment" claim, and since I changed the code and wasn't finished, the user wasn't awarded the expected points.
- the second claim is "Publish Project": since the user didn't have enough points, an exception was thrown by `publish_project` stakeholder but this exception appeared nowhere so I didn't see it
- the test failed without any information at all.

~~Those exceptions should be thrown by `BundlesTest`.~~ A solution should be found to allow developers to see when those problem arise.

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@victornoel it's working as expected: `publish_project` script throws `SoftException` which was catched by `StkSafe` to reply to user with error message. `SoftException` should not fail entire system, they are used to notify claim's author about error. So if user with low reputation will try to publish a project he will see this error: `"Your reputation is 1, while a minimum of 1024 is required in order to do this, see §26" `

**Comment by victornoel** [comment]
@g4s8 I'm talking about tests. I understand how things work for production, but during tests, this is not desired.

In this situation, when we have failing tests, we don't know why they fails: I already gave an example of this in the description of this issue.

Sometimes it is worse: they should be failing but they are not because the tests themselves are bugged.

Here is an example:
- `bootstrap_short_named_project` has 2 claims: one to make payment, one to bootstrap.
- payment is only here so that bootstrap is executed, because the test is about bootstrap
- during my changes, I broke the code and make payment wasn't awarding points: so bootstrapping wasn't executed because it failed in the middle of execution, there wasn't enough point for bootstrapping.
- surprisingly, the test still passes and the assertions in `after` are happy (I don't even understand why)

Basically we have a test that test nothing, and we can't even detect that.

I don't have the perfect solution, but I believe this is a real problem: the test base is weak and devs spend too much time fighting with them instead of taking advantage of them.

**Comment by g4s8** [comment]
@victornoel it is a bug in `bootstrap`, not in `bootstrap_short_named_project` test. This test just check that `bootstrap` can create new project in a catalog.

**Comment by victornoel** [comment]
@g4s8 I am talking about both the bundle `bootstraps_short_named_project`. The test passes even though the claim for bootstraping a project is not processed fully by `bootstrap`.

As proof you can simply remove the claim "Make Payment" in `claims.xml` and the test will still pass. This begs the question as to what is this test testing then, and why is there some code in `bootstrap` to handle channel with names of less than 2 characters...

Anyway, this is not really the subject of this issue: the problem is that as a developer, I can't work properly with errors swallowed like that.

**Comment by g4s8** [comment]
@victornoel I've checked this test, and I see that this test is not valid at all, the asserion will always pass because: project title is set up in `Catalog.add()` as `project.pid()` which is called from `BundleTest` before each test, so even if you remove all claims in `claims.xml` the project with valid title will be placed in `catalog.xml`, so actually this test is not verifying anything. But there is not a bug in `BundleTest`, it's a bug in a concrete test `bootstraps_short_named_project` shich should be fixed.

**Comment by victornoel** [comment]
@g4s8 I know, that's what I am trying to tell you :)

And again: this issue is not about the bug in `bootstraps_short_named_project`, this issue is clearly defined in the description.

**Comment by g4s8** [comment]
@victornoel with current title and description this bug is invalid, because `BundleTest` is working fine, some stakeholder scripts may throw exceptions and it should not affect other stakeholders or entire test, test will fail only if assertion fails.

**Comment by victornoel** [comment]
@g4s8 I don't know what you are expecting of me: the title is quite clear and the problem clearly underlined. I will remove the proposed solution since it doesn't seem to be desired as-is.
I would be glad if the ARC could give an opinion as to how to fix this situation.

**Comment by g4s8** [comment]
@victornoel please see https://github.com/zerocracy/farm/commit/5d7110e83a6287fcada3ecdea386362618ad36db commit: now you will see all soft errors from stakeholders in logs

**Comment by victornoel** [comment]
@g4s8 that's great, thanks, I think we can close this issue now :)

**Comment by 0crat** [comment]
Job `gh:zerocracy/farm#1739` is not assigned, can't get performer
